### DISCUSSION TITLE
Create a URL field for the (API) response

### DISCUSSION THREAD

**Discussion by Peter-Blomqvist** [riok/mapperly#800] [Q&A]
I would like to set an image URL to a field (ImageUrl) in the response.
The only way to do this is to add it after the mapping to the response object (Before / After Map). But then the compiler complaints that the response class doesn't have a an accessible constructor with mappable arguments. But when I add a property called ImageUrl in the DTO, the compiler error is cleared. Why won't it work without the DTO property? And it doesn't matter if I add a MapperIgnoreTarget attribute on the ImageUrl field.
Is there another (better?) way to set the image URL in the response? I would like to have it in the mapper.

**Comment by TimothyMakkison** [comment]
Hey, would it be possible for you to share an example of your issue?

**Reply by Peter-Blomqvist** [reply]
Yes, of course. I'm trying out Mapperly and FastendPoints. It works, although there are some challenges with the mapping. Here it is.
The ArticleAttribute class has many properties, but no ImageUrl property, because I think this belongs to the API layer.
```
public record ArticleAttributeResponse(
	string Language,
	string Group,
	string Name,
	string Abbreviation,
	int SequenceNo,
	string ImageUrl
);

public partial class GetArticleMapper
{
	public ArticleAttributeResponse MapAttribute(ArticleAttribute attribute)
	{
		var response = Map(attribute);
		response = response with { ImageUrl = $"/images/article_attributes/id={attribute.AlternativeId}" };

		return response;
	}
	[MapperIgnoreTarget(nameof(ArticleAttributeResponse.ImageUrl))]
	[MapProperty(nameof(ArticleAttribute.AttributeGroupName), nameof(ArticleAttributeResponse.Group))]
	[MapProperty(nameof(ArticleAttribute.AttributeName), nameof(ArticleAttributeResponse.Name))]
	private partial ArticleAttributeResponse Map(ArticleAttribute attribute);
....
}
```

**Reply by latonz** [reply]
You are ignoring a constructor parameter of the target. Mapperly therefore can't resolve the constructor (since it doesn't know what to assign to `ImageUrl`). You may need to assign a default value in the constructor.

**Reply by Peter-Blomqvist** [reply]
OK! Thank you so much! That fixed it.
